### PR TITLE
docs(config options): explain `prCreation=approval`

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Explain what happens when you set `prCreation` to `approval`

## Context

The current docs list `approval` as a valid `allowedValues`, but lacks a proper explanation.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -3140,6 +3141,8 @@ If there are no checks associated, Renovate will create the PR once 24 hrs have
 
 With prCreation set to `status-success`, Renovate creates the PR only if/ once all tests have passed.
 
+With prCreation set to `approval`, Renovate creates the PR only when approved via the Dependency Dashboard.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:3144]:
```suggestion
With prCreation set to `approval`, Renovate creates the PR only when approved via the Dependency Dashboard.
It does not prevent the creation of the _branch_.
```

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -3128,6 +3128,7 @@ This setting tells Renovate when you would like it to raise PRs:
 - `immediate` (default): Renovate will create PRs immediately after creating the corresponding branch
 - `not-pending`: Renovate will wait until status checks have completed (passed or failed) before raising the PR
 - `status-success`: Renovate won't raise PRs unless tests pass
+- `approval`: Renovate waits until getting approval for the update via the Dependency Dashboard
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:3131]:
```suggestion
- `approval`: Renovate creates branches for updates, but only creates the _PR_ after Dependency Dashboard approval
```

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -3128,6 +3128,7 @@ This setting tells Renovate when you would like it to raise PRs:
...
 
 Renovate defaults to `immediate` but you might want to change this to `not-pending` instead.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:3133]: Do you think we should warn users to only use _one_ of these_:

- `dependencyDashboardApproval=true`, or
- `prCreation=approval`

Its probably pretty confusing when you mix-and-match options. 😄

Comment (rarkins) [reply docs/usage/configuration-options.md:3133]: Yes let's warn them here that it's not compatible with dependencyDashboardApproval=true

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -3123,25 +3123,33 @@ This limit is enforced on a per-repository basis.
...
-With prCreation set to `immediate`, you'll get a Pull Request and possible associated notification right away when a new update is available.
-You'll have to wait until the checks have been performed, before you can decide if you want to merge the PR or not.
-
-With prCreation set to `not-pending`, Renovate creates the PR only once all tests have passed or failed.
+When prCreation is set to `not-pending`, Renovate creates the PR only once all tests have passed or failed.
 When you get the PR notification, you can take action immediately, as you have the full test results.
-If there are no checks associated, Renovate will create the PR once 24 hrs have elapsed since creation of the commit.
+If there are no checks associated, Renovate will create the PR once 24 hours have elapsed since creation of the commit.
+
+When prCreation is set to `status-success`, Renovate creates the PR only if all tests have passed.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:3140]:
```suggestion
When prCreation is set to `status-success`, Renovate creates the PR only if all tests have passed.
When a branch gets "stuck" due to a failing test: select the corresponding PR from the Dependency Dashboard, and push your fixes to the branch.
```

We should probably explain how the users can fix a "stuck branch due to failing test(s)". Here's a draft, feel free to edit the comment and apply the final suggestion. 😉

File: docs/usage/configuration-options.md [hunk 5]
```diff
@@ -3123,25 +3123,34 @@ This limit is enforced on a per-repository basis.
...
 
-This setting tells Renovate when you would like it to raise PRs:
+This setting tells Renovate _when_ to create PRs:
 
-- `immediate` (default): Renovate will create PRs immediately after creating the corresponding branch
-- `not-pending`: Renovate will wait until status checks have completed (passed or failed) before raising the PR
-- `status-success`: Renovate won't raise PRs unless tests pass
+- `immediate` (default): Renovate creates PRs immediately after creating the corresponding branch
+- `not-pending`: Renovate waits until status checks have completed (passed or failed) before raising the PR
+- `status-success`: Renovate only creates PRs when the the test pass
```
Comment (rarkins) [inline docs/usage/configuration-options.md:3130]:
```suggestion
- `status-success`: Renovate only creates PRs if/when the the test pass
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**